src/HealthChecks.AzureServiceBus/AzureEventHubHealthCheck.cs

```diff
@@ -1,46 +1,49 @@
-using Azure.Core;
-using Azure.Messaging.EventHubs;
 using Azure.Messaging.EventHubs.Producer;
+using HealthChecks.AzureServiceBus.Configuration;
 using Microsoft.Extensions.Diagnostics.HealthChecks;
 
 namespace HealthChecks.AzureServiceBus;
 
 public class AzureEventHubHealthCheck : IHealthCheck
 {
     private const string ENTITY_PATH_SEGMENT = "EntityPath=";
-
-    private readonly string? _connectionString;
-    private readonly string? _endpoint;
-    private readonly string? _eventHubName;
-    private readonly TokenCredential? _tokenCredential;
-    private readonly EventHubConnection? _connection;
+    private readonly AzureEventHubHealthCheckOptions _options;
 
     private string? _connectionKey;
 
     private string ConnectionKey =>
-        _connectionKey ??= _connectionString ?? $"{_endpoint}_{_eventHubName}";
+        _connectionKey ??= _options.ConnectionString ?? $"{_options.FullyQualifiedNamespace}_{_options.EventHubName}";
 
-    public AzureEventHubHealthCheck(string connectionString, string eventHubName)
+    public AzureEventHubHealthCheck(AzureEventHubHealthCheckOptions options)
     {
-        Guard.ThrowIfNull(connectionString, true);
-        Guard.ThrowIfNull(eventHubName, true);
+        _options = options;
 
-        _connectionString = connectionString.Contains(ENTITY_PATH_SEGMENT) ? connectionString : $"{connectionString};{ENTITY_PATH_SEGMENT}{eventHubName}";
-    }
+        if (!string.IsNullOrWhiteSpace(options.ConnectionString))
+        {
+            Guard.ThrowIfNull(options.EventHubName, true);
 
-    public AzureEventHubHealthCheck(EventHubConnection connection)
-    {
-        Guard.ThrowIfNull(connection);
+            string connectionString = options.ConnectionString!;
+            if (!connectionString.Contains(ENTITY_PATH_SEGMENT))
+                options.ConnectionString = $"{connectionString};{ENTITY_PATH_SEGMENT}{options.EventHubName}";
```

Comment:
😕 I would not modify the state (at least public) of the outer object.

Reply:
Why do we need this modification here? Does it fix common typo in connection string?

Reply:
> 😕 I would not modify the state (at least public) of the outer object.

Yeah, it felt a bit strange and I remember now that was why I didn't have a field for the option class. What do you suggest?

> Why do we need this modification here? Does it fix common typo in connection string?

I guess because one can provide EventHubName to make it "easier" to use. If it's common or not, I don't know - I don't know what the originator of this code intended.

Reply:
One option is to remove this modification and require "a normal" fully assembled connection string.

Reply:
Then we would need to remove the `eventHubName` and `eventHubNameFactory` parameters from the connectionString overloads of `AddAzureEventHub()` extensions and clarify in the docs how the connection string is intended to be used.

Or we could just move this logic to the extension methods - but that would still change the public property. We could also move it to the option class to handle, however I don't know if I'm ok with "business" logic in an options class.

We could also just do that check in the `CreateClient()`method. I mean, the client is cached anyway.

Reply:
I've pushed new changes for this.